gpytorch/kernels/hamming.py

```diff
@@ -0,0 +1,151 @@
+        beta_constraint: Interval = DEFAULT_BETA_CONSTRAINT,
+    ):
+        super().__init__(batch_shape=batch_shape)
+        self.vocab_size = vocab_size
+        # add alpha (scale) parameter
+        alpha_constraint = Positive() if alpha_constraint is None else alpha_constraint
+        self.register_parameter(
+            name="raw_alpha",
+            parameter=nn.Parameter(torch.zeros(*self.batch_shape, 1)),
+        )
+        if alpha_prior is not None:
+            self.register_prior("alpha_prior", alpha_prior, self._alpha_param, self._alpha_closure)
+        self.register_constraint("raw_alpha", alpha_constraint)
+
+        # add beta parameter
+        beta_constraint = Positive() if beta_constraint is None else beta_constraint
+        self.register_parameter(
+            name="raw_beta",
+            parameter=nn.Parameter(torch.zeros(*self.batch_shape, 1)),
+        )
+        if beta_prior is not None:
+            self.register_prior("beta_prior", beta_prior, self._beta_param, self._beta_closure)
+        self.register_constraint("raw_beta", beta_constraint)
+
+    @property
+    def alpha(self) -> Tensor:
+        return self.raw_alpha_constraint.transform(self.raw_alpha)
+
+    @alpha.setter
+    def alpha(self, value: Tensor):
+        self._set_alpha(value)
+
+    def _alpha_param(self, m: Kernel) -> Tensor:
+        # Used by the alpha_prior
+        return m.alpha
+
+    def _alpha_closure(self, m: Kernel, v: Tensor) -> Tensor:
+        # Used by the alpha_prior
+        return m._set_alpha(v)
+
+    def _set_alpha(self, value: Tensor):
+        # Used by the alpha_prior
+        if not torch.is_tensor(value):
+            value = torch.as_tensor(value).to(self.raw_alpha)
+        self.initialize(raw_alpha=self.raw_alpha_constraint.inverse_transform(value))
+
+    @property
+    def beta(self) -> Tensor:
+        return self.raw_beta_constraint.transform(self.raw_beta)
+
+    @beta.setter
+    def beta(self, value: Tensor):
+        self._set_beta(value)
+
+    def _beta_param(self, m: Kernel) -> Tensor:
+        # Used by the beta_prior
+        return m.beta
+
+    def _beta_closure(self, m: Kernel, v: Tensor) -> Tensor:
+        # Used by the beta_prior
+        return m._set_beta(v)
+
+    def _set_beta(self, value: Tensor):
+        # Used by the beta_prior
+        if not torch.is_tensor(value):
+            value = torch.as_tensor(value).to(self.raw_beta)
+        self.initialize(raw_beta=self.raw_beta_constraint.inverse_transform(value))
+
+    def _imq(self, dist: Tensor) -> Tensor:
+        return ((1 + self.alpha) / (self.alpha + dist)).pow(self.beta)
+
+    def forward(self, x1: Tensor, x2: Tensor, diag: bool = False, **params):
+        # GPyTorch is pretty particular about dimensions so we need to unflatten the one-hot encoding
+        x1 = x1.view(*x1.shape[:-1], -1, self.vocab_size)
+        x2 = x2.view(*x2.shape[:-1], -1, self.vocab_size)
+
+        x1_eq_x2 = torch.equal(x1, x2)
```

Comment:
This is kind of expensive - should we just do `x1 is x2` instead here? I don't really see a lot of situations in which we want to compute the forward pass on the same tensor but that not being the same object.

Reply:
that line was copied from here, should it be changed there as well?
https://github.com/cornellius-gp/gpytorch/blob/master/gpytorch/kernels/kernel.py#L339

Reply:
It looks like the unit tests are failing due to a circular import error, either from the prior or the constraint imports I used for type hints and default args. I can take them out but this behavior seems suboptimal

Reply:
I would advocate for that - @gpleiss, @jacobrgardner, any reason not to do this?

Reply:
Maybe we revisit the x1_eq_x2 question in a second PR where we can make sure it all works?